cmd/serverless-init/cloudservice/cloudrun.go

```diff
@@ -127,3 +155,89 @@ func isCloudRunFunction() bool {
 	log.Debug(fmt.Sprintf("cloud run namespace SET TO: %s", cloudRunFunction))
 	return cloudRunFunctionMode
 }
+
+// GetDefaultConfig returns the medatadata's default config
+func GetDefaultConfig() *GCPConfig {
+	return &GCPConfig{
+		containerIDURL: fmt.Sprintf("%s%s", defaultBaseURL, defaultContainerIDURL),
+		regionURL:      fmt.Sprintf("%s%s", defaultBaseURL, defaultRegionURL),
+		projectIDURL:   fmt.Sprintf("%s%s", defaultBaseURL, defaultProjectID),
+		timeout:        defaultTimeout,
+	}
+}
+
+func getRegion(httpClient *http.Client, config *GCPConfig) string {
+	value := getSingleMetadata(httpClient, config.regionURL)
+	tokens := strings.Split(value, "/")
+	return tokens[len(tokens)-1]
+}
+
+func getSingleMetadata(httpClient *http.Client, url string) string {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		log.Error("unable to build the metadata request, defaulting to unknown")
+		return "unknown"
+	}
+	req.Header.Add("Metadata-Flavor", "Google")
+	res, err := httpClient.Do(req)
+	if err != nil {
+		log.Info("unable to get the requested metadata, defaulting to unknown")
+		return "unknown"
+	}
+	defer res.Body.Close()
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		log.Error("unable to read metadata body, defaulting to unknown")
+		return "unknown"
+	}
+	return strings.ToLower(string(data))
+}
+
+// GetMetaData returns the container's metadata
+func GetMetaData(config *GCPConfig, isCloudRun bool) map[string]string {
+	wg := sync.WaitGroup{}
+	wg.Add(3)
+	mlock := sync.Mutex{}
+	httpClient := &http.Client{
+		Timeout: config.timeout,
+	}
+	metadata := make(map[string]string)
+	go func() {
+		containerID := getSingleMetadata(httpClient, config.containerIDURL)
+		mlock.Lock()
+		metadata["container_id"] = containerID
+		if isCloudRun {
+			metadata[runContainerID] = containerID
+		} else {
+			metadata[functionContainerID] = containerID
+		}
+		mlock.Unlock()
```

Comment:
There's no point in using a goroutine for this work if you're just gonna lock anyway. All these goroutines will be synchronous anyway.